spec/unit/remote_system_spec.rb

```diff
@@ -123,40 +185,87 @@
-        remote_system.retrieve_files(["/foo", "/bar"], "/tmp")
+        rsys_machinery_user.retrieve_files(["/foo", "/bar"], "/tmp")
       end
     end
 
     describe "#read_file" do
       it "retrieves the content of the remote file" do
-        expect(remote_system).to receive(:run_command).and_return("foo")
+        expect(rsys).to receive(:run_command).and_return("foo")
 
-        expect(remote_system.read_file("/foo")).to eq("foo")
+        expect(rsys.read_file("/foo")).to eq("foo")
       end
 
       it "returns nil when the file does not exist" do
         status = double(exitstatus: 1)
-        expect(remote_system).to receive(:run_command).
+        expect(rsys).to receive(:run_command).
           and_raise(Cheetah::ExecutionFailed.new(nil, status, nil, nil))
 
-        expect(remote_system.read_file("/foo")).to be_nil
+        expect(rsys.read_file("/foo")).to be_nil
       end
     end
 
     describe "#inject_file" do
       it "copies a file via scp" do
         expect(Cheetah).to receive(:run).with("scp", "/usr/foobar", "root@remotehost:/tmp")
-        remote_system.inject_file("/usr/foobar", "/tmp")
+        rsys.inject_file("/usr/foobar", "/tmp")
+      end
+
+      it "copies a file via scp with SSH port 5000" do
+        expect(Cheetah).to receive(:run).with("scp", "-P", "5000", any_args)
+        rsys_port.inject_file("/usr/foobar", "/tmp")
+      end
+
+      it "copies a file via scp with a different SSH key" do
+        expect(Cheetah).to receive(:run).with("scp", "-i", "/tmp/private_ssh_key", any_args)
+        rsys_key.inject_file("/usr/foobar", "/tmp")
       end
     end
 
     describe "#remove_file" do
       it "removes a file" do
-        expect(remote_system).to receive(:run_command).with(
+        expect(rsys).to receive(:run_command).with(
           "rm", "/tmp/foo", privileged: true
         )
-        remote_system.remove_file("/tmp/foo")
+        rsys.remove_file("/tmp/foo")
+      end
+    end
+
+    describe "#get_cmd_options" do
+      it "returns nothing when no port and no ssh key was set" do
+        expect(rsys.get_cmd_options).to eq([])
+      end
+
+      it "returns the -p option with its value for ssh" do
+        expect(rsys_port.get_cmd_options(:ssh)).to eq(["-p", "5000"])
+      end
+
+      it "returns the -p option with its value for scp" do
+        expect(rsys_port.get_cmd_options(:scp)).to eq(["-P", "5000"])
+      end
+
+      it "returns the -i option with its value" do
+        expect(rsys_key.get_cmd_options(:ssh)).to eq(["-i", "/tmp/private_ssh_key"])
+      end
+
+      it "raises an exception when no :ssh and no :scp flag was set" do
+        expect { rsys_port.get_cmd_options }.to raise_error(Machinery::Errors::MachineryError)
+      end
+    end
+
+    describe "#build_basic_ssh_command" do
+      it "returns a valid command for ssh" do
+        expect(rsys_port_and_key.build_basic_ssh_command).to eq([
+          "ssh", "-p", "5000", "-i", "/tmp/private_ssh_key"
+        ])
```

Comment:
Indent the right bracket the same as the first position after the preceding left parenthesis.